dissect/target/filesystems/config.py

```diff
@@ -0,0 +1,181 @@
+from __future__ import annotations
+
+import io
+import textwrap
+from logging import getLogger
+from typing import Any, BinaryIO, Iterator, Optional, Union
+
+from dissect.target import Target
+from dissect.target.exceptions import ConfigurationParsingError, FileNotFoundError
+from dissect.target.filesystem import Filesystem, FilesystemEntry, VirtualFilesystem
+from dissect.target.helpers import fsutil
+from dissect.target.helpers.configparser import (
+    ConfigurationParser,
+    ParserOptions,
+    parse,
+)
+
+log = getLogger(__name__)
+
+
+def create_entry(
+    fs: Filesystem,
+    path: str,
+    entry: FilesystemEntry,
+    hint: str,
+    collapse: str,
+    seperator: tuple[str],
+    comment_prefixes: tuple[str],
+) -> Union[FilesystemEntry, ConfigurationParser]:
+    if entry.is_file():
+        options = ParserOptions(collapse, seperator, comment_prefixes)
+        parser_items = parse(entry, hint, options)
+        return ConfigurationEntry(fs, path, entry, parser_items=parser_items)
+    return entry
+
+
+class ConfigurationFilesystem(VirtualFilesystem):
+    __fstype__: str = "META:configuration"
+
+    def __init__(self, target: Target, path: str, **kwargs):
+        super().__init__(**kwargs)
+        self.root.top = target.fs.get(path)
+
+    def _get_till_file(self, path: str, relentry: FilesystemEntry) -> tuple[list[str], FilesystemEntry]:
+        entry = relentry or self.root
+
+        path = fsutil.normalize(path, alt_separator=self.alt_separator).strip("/")
+
+        if not path:
+            return [], entry
+
+        parts = path.split("/")
+
+        for idx, part in enumerate(parts):
+            # Resolve link
+            if entry.is_symlink():
+                entry = entry.readlink_ext()
+
+            if part == ".":
+                continue
+            elif part == "..":
+                entry = entry.up or self.root
+                continue
+
+            if entry is self.root:
+                entry = self.root.top
+
+            entry = entry.get(part)
+            if entry.is_file():
+                break
+
+        return parts[idx:], entry
+
+    def get(
+        self,
+        path: Optional[str] = None,
+        relentry: Optional[FilesystemEntry] = None,
+        hint: Optional[str] = None,
+        collapse: Optional[Union[bool, set]] = None,
+        seperator: Optional[tuple[str]] = None,
+        comment_prefixes: Optional[tuple[str]] = None,
+    ) -> Union[FilesystemEntry, ConfigurationEntry]:
+        parts, entry = self._get_till_file(path or "", relentry)
+
+        for part in parts:
+            if isinstance(entry, ConfigurationEntry):
+                entry = entry.get(part)
+            else:
+                try:
+                    options = parse(entry, hint, collapse, seperator, comment_prefixes)
+                    entry = ConfigurationEntry(self, part, entry, options)
```

Comment:
This seems incorrect?

Also note that a `FilesystemEntry`'s `path` should be a full path, not just a path part